### PR TITLE
ebpf: create function prototypes in all headers

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
commit f2feed0b
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 22 14:09:18 2023

    ebpf: start using eBPF own macros
    
    Use already provided macros for accessing types and fields:
    
    1. allows better type casting instead of relying on (void *).
    2. allows multiple fields accessible in a single step.
    
    Note: This change was made possible after non CO-RE code was dropped.
    
    Fixes: #898

commit ec16789e
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Mon May 22 06:37:03 2023

    ebpf: create function prototypes in all headers
    
    Fixes: #2576

### CODE REVIEW

**Review by AlonZivony — APPROVED** [review]
Looks great :)

**Comment by AlonZivony** [comment]
@rafaeldtinoco, you don't think we should move the functions documentation to the header file?

File: pkg/ebpf/c/common/common.h [hunk 1]
```diff
@@ -7,6 +7,12 @@
 
 #include <maps.h>
 
+// PROTOTYPES
+
+#define sai static __always_inline
```
[the remark this answers is not in the document]
Comment (AlonZivony) [reply pkg/ebpf/c/common/common.h:12]: I am for the change, as we declare all functions like this and I don't really think its important to see it all.
It does make the code less readable for people outside of the project. I don't think it is that hard though.

I do think that `sai` is not so clear, as the initial letters makes sense only after reading the macro.
Maybe change it to something like `def` or `func`, to make it clear that its the standard way to declare functions in our eBPF code?

[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply pkg/ebpf/c/common/common.h:12]: Ok, makes sense for me. I'll try to rename it and we can all agree on the next change (or not, lets see).

**Comment by rafaeldtinoco** [comment]
Integration tests are intermittent. Not only in this PR, I keep failing, from time to time:

image

Its likely related to timeouts when pulling alpine latest image.

**Comment by geyslan** [comment]
> Integration tests are intermittent. Not only in this PR, I keep failing, from time to time:
> 
> Its likely related to timeouts when pulling alpine latest image.

This is already being fixed via #2688, by using timeout per command.

image

**Comment by rafaeldtinoco** [comment]
Alright, tests have passed, and E2E tests have also passed.

File: pkg/ebpf/c/common/common.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply pkg/ebpf/c/common/common.h]: `statfunc` it is then. Looks much better indeed.

[the remark this answers is not in the document]
Comment (rafaeldtinoco) [reply pkg/ebpf/c/common/common.h]: @yanivagman let me know if you're +1 with the rename, please.

**Review by josedonizetti — APPROVED** [review]

**Comment by rafaeldtinoco** [comment]
I've tested this change with E2E tests as well. Thanks Jose!

**Pull request merged**